js/test/test.js

```diff
@@ -107,12 +107,40 @@ if (settings && settings.skip) {
     process.exit ();
 }
 
-//-----------------------------------------------------------------------------
+
+// ### common language specific methods ###
+
+async function runTesterMethod(exchange, methodName, ... args) {
+    return await tests[methodName](exchange, ... args);
+}
+
+function testMethodAvailableForCurrentLang(methodName) {
+    return tests[methodName] !== undefined;
+}
+
+function findValueIndexInArray (arr, value) {
+    return arr.indexOf (value);
+}
+
+function exceptionHint (exc) {
+    return '[' + exc.constructor.name + '] ' + exc.message.slice (0, 200);
+}
+// ### end of language specific common methods ###
+
+// ----------------------------------------------------------------------------
+// ### AUTO-TRANSPILER-START ###
+// ----------------------------------------------------------------------------
 
 async function test (methodName, exchange, ... args) {
-    console.log ('Testing', exchange.id, methodName, '(', ... args, ')');
     if (exchange.has[methodName]) {
-        return await (tests[methodName] (exchange, ... args));
+        if (testMethodAvailableForCurrentLang(methodName)) {
```

Comment:
as some other langs (php /py) doesnt have all methods-tests, we have to use this method to ensure the executing lang has the `methodName` test supported.